roadmap/implementors-guide/guide.md

```diff
@@ -1069,9 +1096,33 @@ Dispatch a `PovFetchSubsystemMessage(relay_parent, candidate_hash, sender)` and
 
 (TODO: send statements to Statement Distribution subsystem, handle shutdown signal from candidate backing subsystem)
 
+### Statement Distribution Subsystem
+
+The statement distribution subsystem sends statements to peer nodes, detects double-voting, and tabulates when a sufficient portion of the validator set has unanimously judged a candidate. When judgment is not unanimous, it escalates the issue to misbehavior arbitration.
+
+Statement Distribution is the only validity module subsystem which has any notion of peer nodes or of our own cryptographic keys. It is responsible for signing statements that we have generated and forwarding them, and for detecting a variety of peer node misbehaviors for reporting to Misbehavior Arbitration. Within the Validity module, it's the main point of contact (via the Overseer) with peer nodes. On receiving a signed statement from a peer, assuming the peer receipt state machine is in an appropriate state, it sends the Candidate Receipt to the Candidate Backing subsystem to double-check the peer's judgment.
+
+#### Peer Recipet State Machine
+
+There is a very simple state machine which governs which messages we are willing to receive from peer validators. Not depicted in the state machine: on initial receipt of any `SignedStatement`, validate that the provided signature does in fact sign the included data.
+
+A: Initial State. Receive `SignedStatement(Statement::Second)`: extract `Statement`, forward to Candidate Backing, proceed to B. Receive any other `SignedStatement` variant: drop it.
```

Comment:
It should be fine to receive a `Valid(X)` signed by A before a `Seconded(Y)`. I can see the justification for not allowing it, but it does not seem to get us much of a win. Requiring that we know `Seconded(Y)` before `Valid(Y)` already restricts us to a bounded set of messages, and is simpler to implement (as it is a more focused condition).

Reply:
Yes, I'll add that this state machine applies for each individual parablock candidate, not for all candidates from a particular validator.

Reply:
And then there should be a small note in there about knowledge tracking for our peers, so we know who knows which candidates based on what they've sent or what we've sent them